### PR TITLE
kraken: replace type.h header by smaller headers for ptref

### PR DESCRIPTION
**Pull request by benoit-bst**
I would like start cleaning **type.h** header inside navitia. 
Nevertheless, it is hard to complete the task in one PR, given the width of the problem. Effectively, it is included 46 times. 

```
#grep results
source/ptreferential/ptref_graph.h|33 col 11| #include "type/type.h"
source/ptreferential/ptreferential_ng.cpp|40 col 11| #include "type/type.h"  //TODO: move static_data
source/equipment/equipment_api.h|33 col 11| #include "type/type.h"
...
source/kraken/tests/disruption_reader_test.cpp|35 col 11| #include "type/type.h"
source/kraken/tests/worker_test.cpp|42 col 11| #include "type/type.h"
source/autocomplete/tests/test.cpp|44 col 11| #include "type/type.h"
source/routing/tests/next_stop_time_test.cpp|39 col 11| #include "type/type.h"
```
This work is the continuity of previous PRs _(Split type.h into several headers and cleaning dependencies_).
To avoid conflicts and increase readibility, I perfer to split in different batch of PRs. 
So. I am focusing on ptref for now :
```
source/ptreferential/ptref_graph.h|33 col 11| #include "type/type.h"
source/ptreferential/ptreferential_ng.cpp|40 col 11| #include "type/type.h"  //TODO: move static_data
```
The main task is to **extract static_data from type.h**

TODO : it remains **44** after that... :hammer:

### CODE REVIEW

File: source/ptreferential/ptref_graph.h [hunk 1]
```diff
@@ -30,9 +30,8 @@ www.navitia.io
 
 #pragma once
 
-#include "type/type.h"
```
Comment (benoit-bst) [inline source/ptreferential/ptref_graph.h:33]: culprit 1

File: source/ptreferential/ptreferential_ng.cpp [hunk 2]
```diff
@@ -37,8 +37,8 @@ www.navitia.io
 #include <boost/spirit/include/qi.hpp>
 #include <boost/spirit/include/phoenix.hpp>
 #include "type/line.h"
-#include "type/type.h"  //TODO: move static_data
```
Comment (benoit-bst) [inline source/ptreferential/ptreferential_ng.cpp:40]: culprit 2

**Comment by benoit-bst** [comment]
Jenkins, test this please !

**Review by kinnou02 — APPROVED** [review]

File: source/ptreferential/ptreferential_ng.cpp [hunk 3]
```diff
@@ -416,7 +416,7 @@ std::string make_request(const Type_e requested_type,
                          const boost::optional<boost::posix_time::ptime>& until,
                          const type::RTLevel rt_level,
                          const type::Data& data) {
-    type::static_data* static_data = type::static_data::get();
+    auto* static_data = navitia::type::static_data::get();
```
Comment (AurelienLP) [inline source/ptreferential/ptreferential_ng.cpp]: really minor,  `const` also maybe ?

File: source/type/static_data.h [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+namespace navitia {
+namespace type {
+
+struct static_data {
+private:
+    static static_data* instance;
+
+public:
+    static static_data* get();
+    // static std::string getListNameByType(Type_e type);
```
Comment (AurelienLP) [inline source/type/static_data.h]: remove ?

**Review by AurelienLP — APPROVED** [review]

**Comment by benoit-bst** [comment]
Jenkins, test this please !

**Pull request merged**